module/VuFind/src/VuFind/View/Helper/Root/SearchTabs.php

```diff
@@ -431,7 +431,15 @@ protected function buildUrlHiddenFilters(Results $results, $filters)
         foreach ($filters as $filter) {
             $params->addHiddenFilter($filter);
         }
-        $urlParams = $results->getUrlQuery()->getParams(false);
-        return $urlParams !== '?' ? $urlParams : '';
+        $urlParams = [];
+        $hiddenFilters = $params->getHiddenFilters();
+        foreach ($hiddenFilters as $field => $values) {
+            foreach ($values as $current) {
+                $urlParams['hiddenFilters'][] = $field . ':"' . $current . '"';
```

Comment:
It's passed on to http_build_query, which should take care of things. But I can see that UrlQueryHelper has its own method `buildQueryString` which creates `hiddenFilters[]=...` instead of `hiddenFilters[0]...` and avoids something that messes up ampersands (not sure what). I've now changed `buildQueryString` to public static and use it instead. What do you think?

Reply:
@demiankatz Ah, I see. I don't think the old code did anything different. And I don't think we want to handle escaping for the backend in the query params. I believe it's more coherent to do that when the backend params are gathered and sent to the backend regardless of whether they came from query params or config or whatever. 

`getHiddenFiltersAsQueryParams` on the other hand sounds like it could work. I'll see how it turns out.

Reply:
@demiankatz, I'm not sure that it's specifically escaping for the backend -- if we have a hidden `field` filter with the value `"value"`, what is the expected query parameter? `field:""field""`? Or `field:"\"field\""`? Maybe we've never encountered this situation and answered that question, but it feels like something that might cause problems one way or another.

Reply:
Right. I'm not sure what this should do. I just parroted the code in UrlQueryHelperFactory without giving it any further thought, but it's obviously something that needs to be checked.

Reply:
I guess my bottom line is that I don't want to create more duplicate instances of the same potentially questionable logic. But if we can refactor things so it only happens in one place, then we can revisit the deeper implications separately in the future. :-)

Reply:
I've now added methods to the base Params class to return the filters. We still have a few other places that build filters, and I've tried to ensure that all of them do the same thing. Perhaps some further refactoring could be done, but I think I'll need to extract the changes into a separate pull request first.

Looks like quotes in a filter don't pose a problem provided that the filter value is enclosed in quotes in the url. `Params::parseFilter` removes the outermost quotes, so ensuring that they are always present will make sure that the actual filter value is untouched. I don't really like this, but there's no way to change parseFilter without introducing serious BC issues.

Reply:
Hidden filter handling now in #2649.